src/Microsoft.AspNet.Mvc.ModelBinding/Internal/DictionaryHelper.cs

```diff
@@ -28,6 +29,16 @@ public static class DictionaryHelper
                     continue;
                 }
 
+                if (key.StartsWith("[", StringComparison.OrdinalIgnoreCase))
```

Comment:
Why was this needed? Was it to support the other scenario of [0].Name etc? I am really not sure why there is an entry like that in MSD. Could you elaborate more on the scenario. Moreover this is a common method which gets used to determine Validation state. I am not sure adding this would not affect that code path.